src/universal-dapp.js

```diff
@@ -18,14 +17,13 @@ function UniversalDApp (contracts, options, transactionDebugger) {
 
   self.web3 = options.web3;
   self.transactionDebugger = transactionDebugger;
+
   if (options.mode === 'vm') {
     // FIXME: use `options.vm` or `self.vm` consistently
     options.vm = true;
 
     self.accounts = {};
-
-    self.vm = new EthJSVM(null, null, { activatePrecompiles: true, enableHomestead: true });
```

Comment:
we need to move the instanciation of the vm away from udapp because it is possible now to switch the provider used by the debugger (web3 or VM) independently of udapp (udapp may not be instanciated when the user switch to the vm provider).
Also as the vm has kind of the same role as web3 maybe it makes sense to wrap the vm into an object which provides the same interface as web3. 
There's certainly room for improvement, but maybe better to address that in an another PR.
switching to an event based env could be a nice opportunity